### DISCUSSION TITLE
Outwards order for expensive cells?

### DISCUSSION THREAD

**Discussion by Izhaki** [TanStack/virtual#309] [Ideas] [answered]
This is a similar feature request to that on https://github.com/bvaughn/react-window/issues/645.

---

We are using the virtualiser to render audio waveforms and their overview takes time to calculate. What's more, this will eventually happen in web workers, meaning asynchronously.

For example, each column is 1000px wide. At zoom 1/32 you have 32000 samples to iterate on, per column. With an overscan of 1, this gives nearly 100K samples to iterate on initial render.

It makes more sense in these scenarios to first render the column that is actually seen, only then do the overscan ones.

In fact, I think this makes sense for all cases.

**Comment by piecyk** [comment, marked as answer]
@Izhaki you can controll the order and what is rendered using the rangeExtractor https://tanstack.com/virtual/v3/docs/api/virtualizer#rangeextractor
